src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -318,12 +318,85 @@ spmv_beta_no_transpose (const KokkosKernels::Experimental::Controls& controls,
 			typename YVector::const_value_type& beta,
 			const YVector& y)
 {
-  typedef typename AMatrix::ordinal_type ordinal_type;
+  typedef typename AMatrix::non_const_ordinal_type ordinal_type;
   typedef typename AMatrix::execution_space execution_space;
 
   if (A.numRows () <= static_cast<ordinal_type> (0)) {
     return;
   }
+#if defined(KOKKOS_ENABLE_SERIAL) 
+  if(std::is_same<execution_space,Kokkos::Serial>::value) {
+    /// serial impl                                                                                         
+    typedef typename AMatrix::non_const_value_type value_type;
+    typedef typename AMatrix::non_const_size_type size_type;
+
+    const size_type *__restrict__ row_map_ptr = A.graph.row_map.data();
+    const ordinal_type *__restrict__ col_idx_ptr = A.graph.entries.data();
+    const value_type *__restrict__ values_ptr = A.values.data();
+
+    typename YVector::non_const_value_type *__restrict__  y_ptr = y.data();
+    typename XVector::const_value_type *__restrict__  x_ptr = x.data();
+
+    const typename YVector::non_const_value_type zero(0);
+    const ordinal_type nrow = A.numRows();
+    if (alpha == zero) {
+      if (dobeta == 0) {
+        memset(y_ptr, 0, sizeof(typename YVector::value_type)*nrow);
+      } else if (dobeta == 1) {
```

Comment:
Why not check for `dobeta == -1`, it would probably be a bit faster to do it with -= operator than with *= operator?
Again not a big deal

Reply:
This is a case with alpha = zero. When beta is any number, it is scaling. -= cannot be used here.

Reply:
Yes, I guess I should have said, `y = -y`, I am actually hoping that the compiler can detect that alpha is equal to `-1` and can optimize this but again I do not think it is critical here. I think `alpha=-1` is a more common case because of residual calculations.

Reply:
See the comment in the below.